src/coreComponents/physicsSolvers/fluidFlow/StencilAccessors.hpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file StencilAccessors.hpp
+ */
+
+#ifndef GEOSX_PHYSICSSOLVERS_FLUIDFLOW_STENCILACCESSORS_HPP_
+#define GEOSX_PHYSICSSOLVERS_FLUIDFLOW_STENCILACCESSORS_HPP_
+
+#include "mesh/ElementRegionManager.hpp"
+#include "codingUtilities/traits.hpp"
+#include "codingUtilities/Utilities.hpp"
+
+#include <tuple>
+
+namespace geosx
+{
+
+/**
+ * @brief A struct to automatically construct and store element view accessors
+ * @struct StencilAccessors
+ * @tparam TRAITS the pack containing the types of the fields
+ */
+template< typename ... TRAITS >
+class StencilAccessors
```

Comment:
@francoishamon I moved this class into a separate file because `ElementRegionManager` is already pretty bloated, and this kind of storage class seems quite specific to the stencil-based physics solvers (maybe `fluidFlow` is too specific a place for it... we can move it later if we want). I also ended up combining `StencilAccessors` and `StencilMaterialAccessors` into one to avoid duplicating the code for accessing the data - it just has two distinct constructors, one for regular data and one for material.